packages/lib-utils/src/k8s/hooks/watch-resource-types.ts

```diff
@@ -0,0 +1,59 @@
+import type {
+  K8sGroupVersionKind,
+  K8sResourceCommon,
+  K8sResourceKindReference,
+  Selector,
+} from '../../types/k8s';
+
+export type WatchK8sResult<R extends K8sResourceCommon | K8sResourceCommon[]> = [
+  R,
```

Comment:
Nit: TS 4+ allows discrete tuple (array) elements to have meaningful names.

See for example `UseResolvedExtensionsResult` in `packages/lib-core/src/store/useResolvedExtensions.ts`

```ts
type UseResolvedExtensionsResult<TExtension extends Extension> = [
  resolvedExtensions: LoadedExtension<ResolvedExtension<TExtension>>[],
  resolved: boolean,
  errors: unknown[],
];
```

Can be done later as an improvement.